src/repr/src/timestamp.rs

```diff
@@ -77,6 +77,11 @@ pub trait TimestampManipulation:
     /// which must only happen if the timestamp is `Timestamp::minimum()`.
     fn step_back(&self) -> Option<Self>;
 
+    /// Retreat a timestamp by the least amount possible such that
+    /// `ts.saturating_step_back().unwrap().less_than(ts)` is true. If not possible (which must only happen if the
+    /// timestamp is `Timestamp::minimum()`), then return the originally given timestamp.
+    fn saturating_step_back(&self) -> Self;
```

Comment:
I think we should either add this method to both the trait and the `Timestamp` type or to neither. I'd prefer not adding it because so far `TimestampManipulation` does not contain any of the `saturating_*`/`checked_*` methods and just adding a single one of them breaks the consistency.

How about we just call `let time = time.step_back().unwrap_or(time.clone());` at the one place we need it?

Reply:
Ok, I've changed the PR.